### PR TITLE
Update/favorite feature

### PR DESCRIPTION
**Pull request by Tkachenko01001**

### CODE REVIEW

File: src/components/item-card/ItemCard.tsx [hunk 1]
```diff
@@ -2,29 +2,31 @@ import styles from './itemCard.module.scss'
...
 
-  const dispatch = useDispatch()
-
   const handleClickFavorite = (
     e: React.MouseEvent<SVGSVGElement, MouseEvent>
   ) => {
     e.preventDefault()
-    dispatch(toggleFavorite(item))
+    const accessToken = getToken()
+    setIsModalOpen(!accessToken && true)
```
Comment (pqoqubbw) [inline src/components/item-card/ItemCard.tsx]:
```suggestion
    setIsModalOpen(!accessToken)
```

File: src/pages/favorite-page/FavoritePage.tsx [hunk 2]
```diff
@@ -1,37 +1,68 @@
...
+  const [isModalOpen, setIsModalOpen] = useState(false)
   const allItem = useSelector((state: RootState) => state.items.items)
+  const accessToken = getToken()
   const favoriteItems = allItem.filter((item) => {
     return item.is_favorite
   })
 
+  const handleClickAuth = () => {
+    const accessToken = getToken()
+    setIsModalOpen(!accessToken && true)
```
Comment (pqoqubbw) [inline src/pages/favorite-page/FavoritePage.tsx]:
```suggestion
    setIsModalOpen(!accessToken)
```

File: src/redux/products/slice.ts [hunk 3]
```diff
@@ -33,20 +34,14 @@ const INITIAL_STATE: ItemsState = {
   allCategories: null,
   offset: 0,
   sortValue: SortValue.lowPrice,
-  selectedWeight: []
+  selectedWeight: [],
+  isFavorite: []
```
Comment (pqoqubbw) [inline src/redux/products/slice.ts:38]: did you add this to types?

**Review by pqoqubbw — CHANGES_REQUESTED** [review]

File: src/redux/products/slice.ts (hunk 3 above)
Comment (Tkachenko01001) [reply src/redux/products/slice.ts:38]: I'll add the types, this is a small preparation for the future

**Pull request merged**